cpp/CMakeLists.txt

```diff
@@ -37,300 +80,235 @@ if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+               CUDA_STANDARD_REQUIRED              ON
+               POSITION_INDEPENDENT_CODE           ON
+               INTERFACE_POSITION_INDEPENDENT_CODE ON
+)
+
+target_compile_options(cuspatial
+            PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${CUSPATIAL_CXX_FLAGS}>"
+                    "$<$<COMPILE_LANGUAGE:CUDA>:${CUSPATIAL_CUDA_FLAGS}>"
+)
+
+target_compile_definitions(cuspatial
+            PUBLIC "$<$<COMPILE_LANGUAGE:CXX>:${CUSPATIAL_CXX_DEFINITIONS}>"
+                   "$<$<COMPILE_LANGUAGE:CUDA>:${CUSPATIAL_CUDA_DEFINITIONS}>"
+)
+
+# Disable Jitify log printing. See https://github.com/NVIDIA/jitify/issues/79
+target_compile_definitions(cuspatial PRIVATE "JITIFY_PRINT_LOG=0")
+
+# Specify include paths for the current target and dependents
+target_include_directories(cuspatial
+           PUBLIC      "$<BUILD_INTERFACE:${CUSPATIAL_SOURCE_DIR}/include>"
+           PRIVATE     "$<BUILD_INTERFACE:${CUSPATIAL_SOURCE_DIR}/src>"
+           INTERFACE   "$<INSTALL_INTERFACE:include>")
+
+# Add Conda library paths if specified
+if(CONDA_LINK_DIRS)
+    target_link_directories(cuspatial PUBLIC "$<BUILD_INTERFACE:${CONDA_LINK_DIRS}>")
+endif()
 
-###################################################################################################
-# - add google benchmark --------------------------------------------------------------------------
+# Add Conda include paths if specified
+if(CONDA_INCLUDE_DIRS)
+    target_include_directories(cuspatial PUBLIC "$<BUILD_INTERFACE:${CONDA_INCLUDE_DIRS}>")
+endif()
 
-if(BUILD_BENCHMARKS)
+# Per-thread default stream
+if(PER_THREAD_DEFAULT_STREAM)
+    target_compile_definitions(cuspatial PUBLIC CUDA_API_PER_THREAD_DEFAULT_STREAM)
+endif()
 
-  include(ConfigureGoogleBenchmark)
+# Disable NVTX if necessary
+if(NOT USE_NVTX)
+    target_compile_definitions(cuspatial PUBLIC NVTX_DISABLE)
+endif()
 
-  if(GBENCH_FOUND)
-    message(STATUS "Google C++ Benchmarking Framework (Google Benchmark) found in ${GBENCH_ROOT}")
-    include_directories(${GBENCH_INCLUDE_DIR})
-    add_subdirectory(${CMAKE_SOURCE_DIR}/benchmarks)
-  else()
-    message(AUTHOR_WARNING "Google C++ Benchmarking Framework (Google Benchmark) not found: automated tests are disabled.")
-  endif(GBENCH_FOUND)
+# Define spdlog level
+target_compile_definitions(cuspatial PUBLIC "SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_${RMM_LOGGING_LEVEL}")
 
-endif(BUILD_BENCHMARKS)
+# Specify the target module library dependencies
+target_link_libraries(cuspatial PUBLIC GDAL::GDAL cudf::cudf)
 
-###################################################################################################
-# - include paths ---------------------------------------------------------------------------------
+if(CUDA_STATIC_RUNTIME)
+    target_link_libraries(cuspatial PUBLIC CUDA::cusparse CUDA::nvrtc CUDA::cudart_static CUDA::cuda_driver)
+else()
+    target_link_libraries(cuspatial PUBLIC CUDA::cusparse CUDA::nvrtc CUDA::cudart CUDA::cuda_driver)
```

Comment:
Do we actually need `cuda_driver` here?

Reply:
no, likely not.

Reply:
Actually, I'd expect linking cuDF would pull these in...